apps/users/models.py

```diff
@@ -352,7 +352,7 @@ def needs_tougher_password(user):
 
     @property
     def source(self):
-        if self.fxa_id:
+        if self.fxa_id or not self.pk:
```

Comment:
It would be great to get a comment here explaining that unsaved users are set to `fxa` to avoid leaking data through the API.

However, this makes it seem like the wrong place to optimize for data leakage. If this method is exposing the true source of the user then it should be None for unsaved accounts. The API view handler should be the place to make a decision to convert None into `fxa`. That's just food for thought -- if this is the easiest way to implement it then a comment will suffice. I just wouldn't want someone to try and use this method thinking they are getting the source of where the user came from.